plasmapy/particles/tests/test_ionization_state.py

```diff
@@ -705,9 +705,7 @@ def test_kappa_isinf_when_not_set(self):
 def test_iteration_with_nested_iterator():
     hydrogen = IonizationState("p+", n_elem=1e20 * u.m**-3, T_e=10 * u.eV)
 
-    i = 0
-    for _, __ in itertools.product(hydrogen, hydrogen):
-        i += 1
+    i = sum(1 for _, __ in itertools.product(hydrogen, hydrogen))
```

Comment:
Isn't this just
```suggestion
    i = len(itertools.product(hydrogen, hydrogen))
```

Reply:
It gave an error that there is `__len__` attribute associated with what's in parentheses.  I guess that's because it's an iterator or generator maybe?  Your suggestion does look significantly cleaner though.

Reply:
Ah, we should have done `len(list(itertools.product` then!